### PR TITLE
Mark functions as static and const

### PR DESCRIPTION
**Pull request by AtariDreams**

### CODE REVIEW

File: include/vcpkg/base/format.h [hunk 1]
```diff
@@ -25,7 +25,7 @@ namespace fmt
     template<>
     struct formatter<vcpkg::LineInfo, char>
     {
-        constexpr auto parse(format_parse_context& ctx) const -> decltype(ctx.begin())
+        static constexpr auto parse(format_parse_context& ctx) -> decltype(ctx.begin())
```
Comment (BillyONeal) [inline include/vcpkg/base/format.h:28]: As I indicated in the other PR, this is following the formatter protocol, so this change damages correctness. https://fmt.dev/latest/api.html#formatting-user-defined-types

(fmt wants to call with `a.b` syntax not `a::b` syntax)

Adding const would be OK.

File: include/vcpkg/base/format.h [hunk 2]
```diff
@@ -49,7 +49,7 @@ namespace fmt
     template<>
     struct formatter<std::error_code, char> : formatter<std::string, char>
     {
-        constexpr auto parse(format_parse_context& ctx) const -> decltype(ctx.begin())
+        static constexpr auto parse(format_parse_context& ctx) -> decltype(ctx.begin())
```
Comment (BillyONeal) [inline include/vcpkg/base/format.h:52]: Ditto formatter correctness damage.

File: include/vcpkg/metrics.h [hunk 3]
```diff
@@ -98,24 +98,24 @@ namespace vcpkg
         Metrics(const Metrics&) = delete;
         Metrics& operator=(const Metrics&) = delete;
 
-        void set_send_metrics(bool should_send_metrics);
-        void set_print_metrics(bool should_print_metrics);
+        static void set_send_metrics(bool should_send_metrics);
```
Comment (BillyONeal) [inline include/vcpkg/metrics.h:101]: https://github.com/microsoft/vcpkg-tool/pull/723 is a better fix

File: include/vcpkg/paragraphparser.h [hunk 4]
```diff
@@ -57,7 +57,7 @@ namespace vcpkg
         void required_field(StringView fieldname, std::pair<std::string&, TextRowCol&> out);
 
         std::string optional_field(StringView fieldname);
-        void optional_field(StringView fieldname, std::pair<std::string&, TextRowCol&> out);
+        void optional_field(StringView fieldname, std::pair<std::string&, TextRowCol&> out) const;
```
Comment (BillyONeal) [inline include/vcpkg/paragraphparser.h:60]: This change is incorrect. `*this` doesn't change, but data `*this` controls does.

(Similarly, `T& vector::operator[](size_t)` doesn't change the vector, but it should not be `const`)

File: src/vcpkg-test/dependencies.cpp [hunk 5]
```diff
@@ -58,7 +58,7 @@ struct MockVersionedPortfileProvider : IVersionedPortfileProvider
 
     SourceControlFileAndLocation& emplace(std::string&& name,
                                           Version&& version,
-                                          VersionScheme scheme = VersionScheme::String)
+                                          VersionScheme scheme = VersionScheme::String) const
```
Comment (BillyONeal) [inline src/vcpkg-test/dependencies.cpp:61]: Incorrect: returns reference to mutable.

File: src/vcpkg-test/expected.cpp [hunk 6]
```diff
@@ -76,7 +76,8 @@ namespace
             ++cr->move_assigns;
             return *this;
         }
-        std::string to_string() const { return "a construct tracker"; }
+
+        static std::string to_string() { return "a construct tracker"; }
```
Comment (BillyONeal) [inline src/vcpkg-test/expected.cpp:80]: This is no longer following the obj.to_string() protocol.

File: src/vcpkg/base/system.process.cpp [hunk 7]
```diff
@@ -544,7 +544,7 @@ namespace vcpkg
 
         friend void swap(ProcessInfo& lhs, ProcessInfo& rhs) noexcept { lhs.swap(rhs); }
 
-        unsigned int wait()
+        unsigned int wait() const
```
Comment (BillyONeal) [inline src/vcpkg/base/system.process.cpp:547]: Not logically const.

File: src/vcpkg/dependencies.cpp [hunk 8]
```diff
@@ -288,11 +288,12 @@ namespace vcpkg
 
             void install(Span<const FeatureSpec> specs, UnsupportedPortAction unsupported_port_action);
             void upgrade(Span<const PackageSpec> specs, UnsupportedPortAction unsupported_port_action);
-            void mark_user_requested(const PackageSpec& spec);
+            void mark_user_requested(const PackageSpec& spec) const;
```
Comment (BillyONeal) [inline src/vcpkg/dependencies.cpp:291]: These names certainly don't sound unmodifying?

File: src/vcpkg/install.cpp [hunk 9]
```diff
@@ -216,8 +216,8 @@ namespace vcpkg
             bool operator()(const std::string& lhs, const std::string& rhs) { return lhs < rhs; }
             bool operator()(const file_pack& lhs, const file_pack& rhs) { return lhs.first < rhs.first; }
 #endif
-            bool operator()(const std::string& lhs, const file_pack& rhs) { return lhs < rhs.first; }
-            bool operator()(const file_pack& lhs, const std::string& rhs) { return lhs.first < rhs; }
+            bool operator()(const std::string& lhs, const file_pack& rhs) const { return lhs < rhs.first; }
```
Comment (BillyONeal) [inline src/vcpkg/install.cpp:219]: The other ones should be as well.

**Review by BillyONeal — CHANGES_REQUESTED** [review]

File: src/vcpkg/install.cpp (hunk 9 above)
Comment (AtariDreams) [reply src/vcpkg/install.cpp:219]: I don't think that compilers but I will try now

**Comment by AtariDreams** [comment]
@BillyONeal Done!

**Review by BillyONeal — APPROVED** [review]

**Comment by BillyONeal** [comment]
Thanks :)

**Pull request merged**